lib/array_equals.rb

```diff
@@ -1,5 +1,31 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
+
 def array_equals(array1, array2)
-  raise NotImplementedError
+  
+  
+  if array1 == nil && array2 == nil 
+    return true
+  elsif array1 == nil && array2 != nil 
```

Comment:
You could also do `if array1.nil? || array2.nil?`  because you know by this point that they're not **both** `nil`.